module/apmhttp/client.go

```diff
@@ -114,6 +116,7 @@ func (r *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 
 	r.setHeaders(req, traceContext, propagateLegacyHeader)
 	resp, err := r.r.RoundTrip(req)
+	r.requestTracer.end()
```

Comment:
The response may not have been fully transferred at this point. I think we should end the requestTracer when we end the `external.http` span: either if RoundTrip returns an error (see below, line 122), or when the wrapped response body is consumed.